Jamfile

```diff
@@ -468,21 +469,23 @@ feature export-extra : off on : composite propagated ;
 
 # this is a trick to get filename paths to targets to become shorter
 # making it possible to build on windows, especially mingw seems particular
+# for release builds, disable optimizations as they bump GCC over the edge of
+# allowed memory usage on travis-ci
 variant test_release
 	: release : <asserts>production <debug-symbols>on
-	  <invariant-checks>full <boost-link>shared
-	  <export-extra>on <debug-iterators>on <boost>source <threading>multi
+	  <invariant-checks>full <boost-link>shared <optimization>off
```

Comment:
I would like to have one build configuration that makes sure release builds still work and pass the tests. why did you have to disable optimizations?